### DISCUSSION TITLE
Will slices override each other?

### DISCUSSION THREAD

**Discussion by Sebastian-Nielsen** [pmndrs/zustand#2512] [Q&A]
Say bearSlice and fishSlice both have a variable `isMeat`, won't it break if you then object-destructure those two slices into the same object? The docs doesn't mention this at all. 

```ts
const useBoundStore = create<BearSlice & FishSlice>()((...a) => ({
  ...createBearSlice(...a),
  ...createFishSlice(...a),
}))
```

**Comment by dai-shi** [comment]
I thought it's obvious, but maybe not. Please feel free to open a PR for your suggestion for clarification.
You might like `zustand-lens` namespacing.

**Reply by Sebastian-Nielsen** [reply]
`zustand-lens` is too unpopular for my liking and also it seems to add a lot of complexity to the code; all this just to solve this simple problem? I dont know ... 

I have an ignorant question: Instead of destructering the entire slice into the store, would it work if you simply did this:
```ts
const useBoundStore = create<{bear: BearSlice} & {fish: FishSlice}>()((...a) => ({
  bear: createBearSlice(...a),
  fish: createFishSlice(...a),
}))
```

I already know the answer, but I just don't understand why that won't work:
![image](https://github.com/pmndrs/zustand/assets/34586104/37e82b6e-d957-4c6c-b5b6-7f2f561a2bc7)

[screenshot of the typescript error](https://gyazo.com/6a9a9bf4701706a3d936e7806113e207)

Thoughts?

**Reply by dai-shi** [reply]
You first want to try if your requirement can be done with JS. If it's possible, then typing it is another beast.

**Reply by dbritto-dev** [reply]
@Sebastian-Nielsen yeah, that's possible. Actually, I'm working on updating docs content to add some stuff like that. I'll share you and example with types.

**Reply by Sebastian-Nielsen** [reply]
> @Sebastian-Nielsen yeah, that's possible. Actually, I'm working on updating docs content to add some stuff like that. I'll share you and example with types.

I am eargerly waiting! Please feel free to share it here as soon as you has something working to show 🙏

**Reply by KamaniBhavin** [reply]
I am looking for a solution to this

**Reply by dai-shi** [reply]
BTW, take a look at https://github.com/zustandjs/zustand-slices if you haven't.

**Reply by KamaniBhavin** [reply]
@dai-shi I'm presently addressing this with the solution outlined here: [GitHub Discussion](https://github.com/pmndrs/zustand/discussions/2518#discussioncomment-9269751), and I'll also explore [zustand-slices](https://github.com/zustandjs/zustand-slices).
